### PR TITLE
Update @reach/dialog in pcln-modal for DS v2

### PR DESCRIPTION
**Pull request by jes708**
- Adds an export to core to allow compatibility with pcln-modal v3
- Makes react-popper a peerDependency to alleviate publishing error
- Uses latest @reach/dialog in pcln-modal

### CODE REVIEW

File: packages/modal/package.json [hunk 1]
```diff
@@ -11,31 +11,31 @@
...
-  "dependencies": {
-    "@reach/dialog": "^0.1.2",
-    "prop-types": "^15.7.2",
-    "react-transition-group": "^2.5.3"
-  },
   "peerDependencies": {
-    "pcln-design-system": ">=2.0.0",
+    "@reach/dialog": ">=0.x",
+    "pcln-design-system": ">=2.x",
+    "prop-types": "^15.7.2",
```
Comment (BeniCheni) [inline packages/modal/package.json]: Loosen?

```suggestion
    "prop-types": ">=15.7.2",
```

File: packages/modal/package.json [hunk 2]
```diff
@@ -11,31 +11,31 @@
...
     "react": "^16.5.2",
     "react-dom": "^16.5.2",
+    "react-transition-group": "^2.5.3",
```
Comment (BeniCheni) [inline packages/modal/package.json]:
```suggestion
    "react-transition-group": ">=2.5.3",
```

**Comment by jes708** [comment]
Thanks for reviewing @BeniCheni! Your suggestions have been applied.

**Comment by BeniCheni** [comment]
Thanks, @jes708. If single commit is desire instead of 4 commits in the PR, you can do an interactive rebase to fix up to have beautiful merge:

```
git rebase -i HEAD~5
```
Note: 5 is 4 of your commits + the recent commit in `master`, so you can have a clear outline to fix up the last 3 commits you like

```
git push --force-with-lease origin feature/modal-peers-v2
```

Safe force push to the PR and you would see the beautiful single commit done upon merging. Let me know if you'd like to take merges for your PRs to show some examples.

**Comment by achoboter** [comment]
> Thanks, @jes708. If single commit is desired instead of 4 commits in the PR, you can do an interactive rebase to fix up to have beautiful merge:
> 
> ```
> git rebase -i HEAD~5
> ```
> 
> Note: 5 is 4 of your commits + the recent commit in `master`, so you can have a clear outline to fix up the last 3 commits you like
> 
> ```
> git push --force-with-lease origin feature/modal-peers-v2
> ```
> 
> Safe force push to the PR and you would see the beautiful single commit done upon merging. Let me know if you'd like to take merges for your PRs to show some examples.

You can also just use *Squash and merge* when merging this PR ;)

**Comment by BeniCheni** [comment]
Squashed commits will also show the meaningless commits though.

**Comment by achoboter** [comment]
> Squashed log will also show the meaningless commits though.

I don't believe so... Here is a PR I recently merged with **_Squash and merge_** https://github.com/priceline/design-system/pull/675/commits, it had two commits.

The history for master only shows one commit:
Screen Shot 2020-01-03 at 2 29 32 PM, and the commit log is clean https://github.com/priceline/design-system/commit/3973d9f2b1bdf2a9a0ddd26f774a240232096eda.

**Comment by BeniCheni** [comment]
You are correct about `git log --oneline` format. But I am referring to the granularity of commit detailed body though, no? I am good with --oneline format with just clicking the button with "Squash and merge", but when debugging I tend to look at the granular trace and understand.

Screen Shot 2020-01-03 at 2 39 56 PM

**Pull request merged**